src/COMCorrection.jl

```diff
@@ -0,0 +1,67 @@
+"""
+    COM_correction_function(r,box_sizes, N, original::Bool=false)
+
+Updates the positions by applying a COM correction.
+
+# Arguments
+    - `r`: An array containing all saved particle positions
+    - `box_sizes`: A tuple containing the sizes of the simulation box
+    - `N`: The number of particles.
+    - `original::Bool=false`: Whether to reconstruct the original trajectories.
+
+# Returns
+    - `updated_r_array`: An array containing all updated positions.
+    - if original = true: 
+    `updated_r_array_original`: An array containing all updated positions with reconstructed original trajectories.
+"""
+
+
+function COM_correction_function(r, box_sizes, N, original)
+    # first apply COM correction on original, then convert back to normal by applying pbc?
+    # required to avoid issues where particles end up outside the box!
+    r_original = find_original_trajectories(r, box_sizes)
+
+    # find COM
+    first_x_positions = r_original[1,:,1]
+    first_COM_x = sum(first_x_positions)/N
+
+    first_y_positions = r_original[2,:,1]
+    first_COM_y = sum(first_y_positions)/N
```

Comment:
This only works in 2d. Perhaps it is better to make it dimension independent

```suggestion
    N = size(r, 2)
    first_COM = sum(r_original[:, :, 1], dims=2) ./ N

```